github/Requester.py

```diff
@@ -349,6 +362,41 @@ def __init__(
         self.__userAgent = user_agent
         self.__verify = verify
 
+    def _must_refresh_token(self) -> bool:
+        """Check if it is time to refresh the API token gotten from the GitHub app installation"""
+        if not self.__installation_authorization:
+            return False
+        return (
+            self.__installation_authorization.expires_at
+            < datetime.datetime.utcnow()
+            + datetime.timedelta(seconds=ACCESS_TOKEN_REFRESH_THRESHOLD_SECONDS)
+        )
+
+    def _get_installation_authorization(self):
+        assert self.__app_auth is not None
+        integration = GithubIntegration.GithubIntegration(
+            self.__app_auth.app_id, self.__app_auth.private_key
+        )
+        return integration.get_access_token(
+            self.__app_auth.installation_id,
+            permissions=self.__app_auth.token_permissions,
+        )
+
+    def _refresh_token_if_needed(self) -> None:
+        """Get a new access token from the GitHub app installation if the one we have is about to expire"""
+        if not self.__installation_authorization:
+            return
+        with self.__auth_lock:
+            if self._must_refresh_token():
+                logging.debug("Refreshing access token")
+                self._refresh_token()
+
+    def _refresh_token(self) -> None:
+        """In the context of a GitHub app, refresh the access token"""
+        assert self.__app_auth is not None
```

Comment:
It does not look like this method really requires `self.__app_auth is not None`, it is sufficient that `_get_installation_authorization` has that assertion. This method should not care, it is internal to `_get_installation_authorization`.

Reply:
Indeed, removed